File: server/models/Question.py
from pydantic import BaseModel
from typing import Dict, List
from datetime import datetime

class Question(BaseModel):
    id: int
    category: str
    text: str
    correctAnswer: str
    incorrectAnswers: str
    createdAt: datetime
    answers: List[Dict]

class QuestionDto(BaseModel):
    category: str
    text: str
    correctAnswer: str
    incorrectAnswers: str

File: client/src/types/Question.ts
import type Quiz from "./Quiz";

export interface BaseQuestion {
  id: number;
  category: string;
  text: string;
  correctAnswer: string;
  incorrectAnswers: string;
  quizId?: number;
  quiz?: Quiz;
  question?: string; // Alias for text
}

export interface QuestionWithIndex extends BaseQuestion {
  index: number;
}

export default interface Question extends BaseQuestion {
  index?: number; // For tracking question position in the quiz
  selectedAnswer?: string; // User's selected answer
}